school/src/main/java/com/hogwarts/school/service/StudentService.java

```diff
@@ -1,46 +1,79 @@
 package com.hogwarts.school.service;
 
+import com.hogwarts.school.exception.FacultyNotFoundException;
+import com.hogwarts.school.exception.StudentNotFoundException;
+
+import com.hogwarts.school.model.Faculty;
 import com.hogwarts.school.model.Student;
+import com.hogwarts.school.repositories.FacultyRepository;
+import com.hogwarts.school.repositories.StudentRepository;
 import org.springframework.stereotype.Service;
 
 import java.util.*;
-import java.util.stream.Collectors;
+
 
 @Service
 public class StudentService {
-    private final Map<Long, Student> studentMap = new HashMap<>();
-    private Long generateId = 1L;
 
-    public Student addStudent(Student student) {
-        student.setId(generateId);
-        studentMap.put(student.getId(), student);
-        generateId++;
-        return student;
-    }
+    private final StudentRepository studentRepository;
+    private final FacultyRepository facultyRepository;
 
-    public Student getStudent(Long studentId) {
-        return studentMap.get(studentId);
+    public StudentService(StudentRepository studentRepository,
+                          FacultyRepository facultyRepository) {
+        this.studentRepository = studentRepository;
+        this.facultyRepository = facultyRepository;
     }
 
+    public Student createStudent(Student student) {
+        student.setId(null);
+       fillFaculty(student.getFaculty(),student);
+        return studentRepository.save(student);
+    }
 
-    public Student updateStudent(Student student) {
-        if (!studentMap.containsKey(student.getId())) {
-            return null;
+    public Student findStudent(Long studentId) {
+        Student findStudent = studentRepository.findById(studentId).orElse(null);
+        if (findStudent == null) {
+            throw new StudentNotFoundException();
         }
-        studentMap.put(student.getId(), student);
-        return student;
+        return findStudent;
+    }
+
+
+    public Student updateStudent(Long id, Student student) {
+        Student old = findStudent(id);
+        fillFaculty(old.getFaculty(),student);
```

Comment:
```suggestion
        fillFaculty(student.getFaculty(), old);
```
первым параметром идёт факультет из ЗАПРОСА, вторым параметром - студент из БД (или куда нужно засетить факультет)

Reply:
по-прежнему актуально